src/mono/wasm/runtime/blazor/_Integration.ts

```diff
@@ -0,0 +1,220 @@
+
+    // it would not `loadResource` on types for which there is no typesMap mapping
+    const downloadResource = (asset: AssetEntry): LoadingResource | undefined => {
+        // GOTCHA: the mapping to blazor asset type may not cover all mono owned asset types in the future in which case:
+        // A) we may need to add such asset types to the mapping and to WebAssemblyBootResourceType
+        // B) or we could add generic "runtime" type to WebAssemblyBootResourceType as fallback
+        // C) or we could return `undefined` and let the runtime to load the asset. In which case the progress will not be reported on it and blazor will not be able to cache it.
+        const type = monoToBlazorAssetTypeMap[asset.behavior];
+        if (type !== undefined) {
+            const res = resourceLoader.loadResource(asset.name, asset.resolvedUrl!, asset.hash!, type);
+            asset.pendingDownload = res;
+
+            totalResources++;
+            res.response.then(() => {
+                resourcesLoaded++;
+                if (Module.onDownloadResourceProgress)
+                    Module.onDownloadResourceProgress(resourcesLoaded, totalResources);
+            });
+
+            return res;
+        }
+        return undefined;
+    };
+
+    Module.downloadResource = downloadResource;
+    Module.disableDotnet6Compatibility = false;
+
+    // any runtime owned assets, with proper behavior already set
+    for (const name in resources.runtimeAssets) {
+        const asset = resources.runtimeAssets[name] as AssetEntry;
+        asset.name = name;
+        asset.resolvedUrl = `_framework/${name}`;
+        assets.push(asset);
+        if (asset.behavior === "dotnetwasm") {
+            downloadResource(asset);
+        }
+    }
+    for (const name in resources.assembly) {
+        const asset: AssetEntry = {
+            name,
+            resolvedUrl: `_framework/${name}`,
+            hash: resources.assembly[name],
+            behavior: "assembly",
+        };
+        assets.push(asset);
+        downloadResource(asset);
+    }
+    if (hasDebuggingEnabled(resourceLoader.bootConfig) && resources.pdb) {
+        for (const name in resources.pdb) {
+            const asset: AssetEntry = {
+                name,
+                resolvedUrl: `_framework/${name}`,
+                hash: resources.pdb[name],
+                behavior: "pdb",
+            };
+            assets.push(asset);
+            downloadResource(asset);
+        }
+    }
+    const applicationCulture = resourceLoader.startOptions.applicationCulture || (navigator.languages && navigator.languages[0]);
+    const icuDataResourceName = getICUResourceName(resourceLoader.bootConfig, applicationCulture);
+    let hasIcuData = false;
+    for (const name in resources.runtime) {
+        const behavior = behaviorByName(name) as any;
+        if (behavior === "icu") {
+            if (resourceLoader.bootConfig.icuDataMode === ICUDataMode.Invariant) {
+                continue;
```

Comment:
If mode is invariant we wouldn't like to `assets.push(assset)`, finish early:
```suggestion
                break;
```
Or even better: skip entering the loop if `(resourceLoader.bootConfig.icuDataMode === ICUDataMode.Invariant)`.

Reply:
The loop is over all resources, it can contain also `js-module-threads`

Reply:
Never mind, read it wrong.